### PR TITLE
Added Platform Target Check

### PR DESCRIPTION
**Pull request by nialldonnellyfh**
Added A Check To FHC to verify that the platform target is at least a certain version.

### CODE REVIEW

**Comment by fheng-build-servers** [comment]
@maleck13 @cianclarke please review this PR.

File: lib/cmd/common/version.js [hunk 1]
```diff
@@ -95,5 +95,25 @@ module.exports = {
...
+
+      try{
+        platformDetails = JSON.parse(body);
+      } catch(e){
+        return cb(new Error("Invalid Platform Data"));
+      }
+
+      //If the platform is less that a certain version, then it can't be used with this version of fhc.
+      if(semver.lt(platformDetails.Release, fhc._minPlatformVersion)){
+        return cb(new Error("The Platform You Are Targeting Is Not Compatible With This Version Of fhc. Please Install A Previous Version"));
```
Comment (cianclarke) [inline lib/cmd/common/version.js]: It's be nice if we could call out a specific older point release - maybe parse the semver, then
`"please try installing the previous version: npm install " + semver.major + ".x.x
"

File: package.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
   "name": "fh-fhc",
   "description": "A Command Line Interface for FeedHenry",
   "version": "2.0.0-BUILD-NUMBER",
+  "minPlatformVersion": "7.15.0-1661",
```
Comment (cianclarke) [inline package.json]: Does package.json allow arbitrary data in here?
Is there a property under which we can put this?

**Comment by cianclarke** [comment]
Reviewed so far!

File: package.json (hunk 2 above)
Comment (nialldonnellyfh) [inline package.json]: Might be worth adding an underscore

http://stackoverflow.com/questions/10065564/add-custom-metadata-or-config-to-package-json-is-it-valid. 

Not sure where else I could put it. It's own file maybe?

**Comment by cianclarke** [comment]
lgtm

**Pull request merged**